src/video/wayland/SDL_waylandevents.c

```diff
@@ -2724,6 +2724,13 @@ void Wayland_display_destroy_input(SDL_VideoData *d)
         SDL_free(input->data_device);
     }
 
+    if (input->primary_selection_device != NULL) {
+        if (input->primary_selection_device->selection_offer != NULL) {
+            Wayland_primary_selection_offer_destroy(input->primary_selection_device->selection_offer);
+        }
+        SDL_free(input->primary_selection_device);
+    }
+
```

Comment:
As I was running a test program for these changes I got fsan warnings about primary_selection_device leaking memory at exit. This fixed those warnings.